### PR TITLE
Refactor functions emulator to offload all process related logic to the worker class.

### PR DESCRIPTION
**Pull request by taeold**
Small refactor to move all process management logic from `FunctionsEmulator` to `FunctionsWorker`. I want to start moving all code pieces involved in management of a function process from the functions emulator (which I consider as control plane) to the worker (which I consider to be the data plane).

cc/ @joehan For visibility. Should not impact extensions emulator in any way.

### CODE REVIEW

**Comment by taeold** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4903** <a href="https://app.graphite.dev/github/pr/firebase/firebase-tools/4903" target="_blank">Graphite</a>  👈
    * **PR #4861** <a href="https://app.graphite.dev/github/pr/firebase/firebase-tools/4861" target="_blank">Graphite</a> 
      * **PR #4867** <a href="https://app.graphite.dev/github/pr/firebase/firebase-tools/4867" target="_blank">Graphite</a> 
        * **PR #4886** <a href="https://app.graphite.dev/github/pr/firebase/firebase-tools/4886" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/firebase/firebase-tools/4903?utm_source=stack-comment).

File: src/emulator/functionsRuntimeWorker.ts [hunk 1]
```diff
@@ -47,24 +48,63 @@ export class RuntimeWorker {
...
       this.log("exited");
       this.state = RuntimeWorkerState.FINISHED;
     });
   }
 
+  private processStream(s: Serializable, buf: string): string {
+    buf += s.toString();
+
+    const lines = buf.split("\n");
+    if (lines.length > 1) {
```
Comment (TheIronDev) [inline src/emulator/functionsRuntimeWorker.ts:86]: `> 1` is redundant

**Review by TheIronDev — APPROVED** [review]
LGTM

File: src/emulator/functionsRuntimeWorker.ts [hunk 2]
```diff
@@ -47,24 +48,63 @@ export class RuntimeWorker {
...
+      // slice(0, -1) returns all elements but the last
+      lines.slice(0, -1).forEach((line: string) => {
```
Comment (colerogers) [inline src/emulator/functionsRuntimeWorker.ts:88]: I know this is just copied code, but wanted to call out that we prefer `for(...)` over `forEach(...)`

Comment (taeold) [reply src/emulator/functionsRuntimeWorker.ts:88]: Good new is that I throw away all this code in later PR! Will leave them unchanged in the PR.

File: src/emulator/functionsRuntimeWorker.ts (hunk 1 above)
Comment (taeold) [reply src/emulator/functionsRuntimeWorker.ts:86]: Good new is that I throw away all this code in later PR! Will leave them unchanged in the PR.

**Pull request merged**